### PR TITLE
feat: allow additionalCommandArgs to be passed to barman-cloud-wal-archive

### LINKED ISSUE #4357: [Feature]: Expose additional command args for barman-cloud-wal-archive
### Is there an existing issue already for this feature request/idea?

- [X] I have searched for an existing issue, and could not find anything. I believe this is a new feature request to be evaluated.

### What problem is this feature going to solve? Why should it be added?

To allow access to barman-cloud-wal-archive arguments that are not already exposed in the WalBackupConfiguration. In my case I need access to the sse-kms-key-id argument. 

### Describe the solution you'd like

Add an "additionalCommandArgs" field to WalBackupConfiguration, the same as what there is for DataBackupConfiguration.

### Describe alternatives you've considered

n/a

### Additional context

_No response_

### Backport?

No

### Are you willing to actively contribute to this feature?

Yes

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by pselliotc**
Closes #4357

Adds the additionalCommandArgs property to WalBackupConfiguration, allowing command line args to be passed to barman-cloud-wal-archive.

### CODE REVIEW

**Comment by mnencia** [comment]
Hi @pselliotc, thanks for the contribution. Could you please update `docs/src/backup_barmanobjectstore.md` to include the new parameter?

File: api/v1/cluster_types.go [hunk 1]
```diff
@@ -2333,20 +2348,30 @@ type ExternalCluster struct {
...
+func (cfg *DataBackupConfiguration) AppendAdditionalCommandArgs(options []string) []string {
+	if cfg == nil {
 		return options
 	}
+	return appendAdditionalCommandArgs(cfg.AdditionalCommandArgs, options)
+}
 
-	for _, userOption := range cfg.Data.AdditionalCommandArgs {
+// AppendAdditionalCommandArgs adds custom arguments as barman-cloud-wal-archive command-line options
+func (cfg *WalBackupConfiguration) AppendAdditionalCommandArgs(options []string) []string {
```
Comment (gabriele-wolfox) [inline api/v1/cluster_types.go:2360]: @pselliotc PR looks good to me.
Do you mind if I ask you to add some unit test cases for this method `AppendAdditionalCommandArgs` inside `cluster_types_test.go`?
Same for `appendAdditionalCommandArgs`.

Comment (pselliotc) [reply api/v1/cluster_types.go:2360]: Yes of course! I've added unit tests for both `AppendAdditionalCommandArgs` methods and the private `appendAdditionalCommandArgs` method.

File: api/v1/cluster_types.go [hunk 2]
```diff
@@ -2333,20 +2348,37 @@ type ExternalCluster struct {
...
+	if cfg == nil {
 		return options
 	}
+	return appendAdditionalCommandArgs(cfg.AdditionalCommandArgs, options)
+}
 
-	for _, userOption := range cfg.Data.AdditionalCommandArgs {
+func appendAdditionalCommandArgs(additionalCommandArgs []string, options []string) []string {
+	optionsKeys := map[string]bool{}
```
Comment (gabriele-wolfox) [inline api/v1/cluster_types.go:2368]: If you don't mind, I would like rename `optionsKeys` --> `optionKeys` and `userOption` --> `additionalCommandArg`.

File: api/v1/cluster_types.go [hunk 3]
```diff
@@ -2344,20 +2359,37 @@ type ExternalCluster struct {
...
+	optionKeys := map[string]bool{}
+	for _, option := range options {
+		key := strings.Split(option, "=")[0]
+		if key != "" {
+			optionKeys[key] = true
+		}
+	}
+	for _, additionalCommandArg := range additionalCommandArgs {
+		key := strings.Split(additionalCommandArg, "=")[0]
+		if key == "" || slices.Contains(options, key) || optionKeys[key] {
```
[the remark this answers is not in the document]
Comment (litaocdl) [reply api/v1/cluster_types.go:2407]: you are right, it is needed.

**Review by litaocdl — APPROVED** [review]

**Review by litaocdl — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

**Pull request merged**